### DISCUSSION TITLE
It is not possible to specify the modulusLength for generated RSA keys since the update to 3.x.x

### DISCUSSION THREAD

**Discussion by dlavrenuek** [panva/jose#121] [Ideas]
**Describe the bug**

Jose version 3.x introduced a lot of code changes together with removed functionality that was present in jose 2.x. One of these is setting the `modulusLength` for generated RSA keys. This value is now hardcoded in https://github.com/panva/jose/blob/master/src/runtime/node/generate.ts#L55 and can not be changed during runtime

**To Reproduce**

Code to reproduce the behaviour:

Working code in jose 2.0.3

```js
const key = JWK.generateSync('RSA', 4096);
```

Analogue in jose 3.1.3 with missing option of the modulusLength:

```js
const key = await generate('RS256'); // <-- it is not possible to set the modulusLength
```

- `jose` version: v3.x.x
- affected runtime is: Node any

**Additional context**

- [x] the bug is happening on latest `jose` too.
- [x] i have searched the issues tracker on github for similar issues and couldn't find anything related.

**Comment by panva** [comment]
Hi @dlavrenuek 

This is not appropriate for a bug type issue. It was a conscious decision on my end to narrow the public API surface.

I'll close this for now and consider adding this in a future release. Until then, for Node.js runtime this module is a simple wrapper around promisified [`crypto.generateKeyPair`](https://nodejs.org/api/crypto.html#crypto_crypto_generatekeypair_type_options_callback). If you know the algorithm leads to a given key type, you might as well use the node's builtin.

**Comment by panva** [comment]
added in [v3.2.0](https://github.com/panva/jose/releases/tag/v3.2.0)

**Comment by panva** [comment]
Please consider [supporting](https://github.com/panva/jose?sponsor=1) the library if it provides value to you or your company and this support was of help to you. Supporting the library means, amongst other things, that such support will be available in the future.
